iceoryx2-bb/posix/src/process_state.rs

```diff
@@ -0,0 +1,595 @@
+    ProcessMonitorStateError
+  entry:
+    CorruptedState,
+    Interrupt,
+    UnknownError(i32)
+
+  mapping:
+    ProcessMonitorCreateError
+}
+
+/// A guard for a process that makes the process monitorable by a [`ProcessMonitor`] as long as it
+/// is in scope. When it goes out of scope the process is no longer monitorable.
+///
+/// ```
+/// use iceoryx2_bb_posix::process_state::*;
+///
+/// let process_state_path = FilePath::new(b"process_state_file").unwrap();
+///
+/// // remove potentially uncleaned process state file that can remain when this process crashed
+/// // before
+/// //
+/// // # Safety
+/// //  * This process owns the state file, therefore it is safe to remove it.
+/// unsafe { ProcessGuard::remove(&process_state_path).expect("") };
+///
+/// // start monitoring from this point on
+/// let guard = ProcessGuard::new(&process_state_path).expect("");
+///
+/// // normal application code
+///
+/// // stop monitoring
+/// drop(guard);
+/// ```
+#[derive(Debug)]
+pub struct ProcessGuard {
+    file: Option<File>,
+}
+
+impl Drop for ProcessGuard {
+    fn drop(&mut self) {
+        let path = *self.path();
+        match self.file.take() {
+            Some(f) => match f.remove_self() {
+                Ok(true) => {
+                    trace!(from "ProcessGuard::drop()", "remove process state \"{}\" for monitoring", path);
+                }
+                Ok(false) => {
+                    warn!(from self, "Someone else already removed the owned ProcessGuard file.")
+                }
+                Err(v) => {
+                    error!(from self, "Unable to remove the owned ProcessGuard file ({:?}). This can cause a process to be identified as dead despite it shut down correctly.", v);
+                }
+            },
+            None => {
+                fatal_panic!(from self, "This should never happen! The ProcessGuard contains an empty file!");
+            }
+        }
+    }
+}
+
+const INIT_PERMISSION: Permission = Permission::OWNER_WRITE;
+const FINAL_PERMISSION: Permission = Permission::OWNER_ALL;
+impl ProcessGuard {
+    /// Creates a new [`ProcessGuard`]. As soon as it is created successfully another process can
+    /// monitor the state of the process. One cannot create multiple [`ProcessGuard`]s that use the
+    /// same `path`. But one can create multiple [`ProcessGuard`]s that are using different
+    /// `path`s.
+    ///
+    /// ```
+    /// use iceoryx2_bb_posix::process_state::*;
+    ///
+    /// let process_state_path = FilePath::new(b"process_state_file").unwrap();
+    ///
+    /// // start monitoring from this point on
+    /// let guard = ProcessGuard::new(&process_state_path).expect("");
+    /// ```
+    pub fn new(path: &FilePath) -> Result<Self, ProcessGuardCreateError> {
+        let origin = "ProcessGuard::new()";
+        let msg = format!(
+            "Unable to create new ProcessGuard with the file \"{}\"",
+            path
+        );
```

Comment:
I am also a bit cautious to use this but I am pragmatic when I know that those constructs are not used in hot loops.

But it would be nice to have a performance degradation test somewhere in the pipeline.